### PR TITLE
[Feature][EDT-1648] Disable / enable actions execution

### PR DESCRIPTION
**Pull request by evan-mcgeek**
This PR exposes methods to enable / disable actions exection

## PR Guidelines

- [x] I have read the [contribution and PR guidelines](/chili-publish/studio-sdk/blob/main/CONTRIBUTING.md)

## Related tickets

-   [EDT-1648](https://chilipublishintranet.atlassian.net/browse/EDT-1648)

[EDT-1648]: https://chilipublishintranet.atlassian.net/browse/EDT-1648?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by Dvergar — COMMENTED** [review]
I'm wondering if we should add a note on the fact that a user can (and probably should) use that call before a document is being loaded 🤔

File: packages/sdk/src/controllers/ActionController.ts [hunk 1]
```diff
@@ -139,4 +139,22 @@ export class ActionController {
     setTypeError = async (id: string, hasTypeErrors: boolean) => {
         return this.update(id, { hasTypeError: hasTypeErrors });
     };
+
+    /**
+     * This method disables actions execution on the engine side
```
Comment (Matthiee) [inline packages/sdk/src/controllers/ActionController.ts:144]:
```suggestion
     * This method disables the execution of Actions.
     * Note that any Action will be ignored and forgotten.
```

File: packages/sdk/src/controllers/ActionController.ts [hunk 2]
```diff
@@ -139,4 +139,22 @@ export class ActionController {
...
+     * @returns
+     */
+    disableExecution = async () => {
+        const res = await this.#editorAPI;
+        return res.disableActions().then((result) => getEditorResponseData<null>(result));
+    };
+
+    /**
+     * This method enables actions execution on the engine side
```
Comment (Matthiee) [inline packages/sdk/src/controllers/ActionController.ts:153]:
```suggestion
     * This method enables the execution of Actions.
     * Note this is enabled by default.
```

File: packages/sdk/src/controllers/ActionController.ts [hunk 3]
```diff
@@ -1,6 +1,6 @@
-import { ActionDeltaUpdate, ActionTrigger, DocumentAction } from '../types/ActionTypes';
-import { EditorAPI, Id } from '../types/CommonTypes';
-import { getEditorResponseData } from '../utils/EditorResponseData';
+import {ActionDeltaUpdate, ActionTrigger, DocumentAction} from '../types/ActionTypes';
+import {EditorAPI, Id} from '../types/CommonTypes';
+import {getEditorResponseData} from '../utils/EditorResponseData';
```
Comment (Matthiee) [inline packages/sdk/src/controllers/ActionController.ts:3]: Fix formatting

File: packages/sdk/src/tests/controllers/ActionController.test.ts [hunk 4]
```diff
@@ -1,7 +1,7 @@
-import { ActionDeltaUpdate, ActionEditorEvent, ActionTrigger } from '../../types/ActionTypes';
-import { EditorAPI } from '../../types/CommonTypes';
-import { ActionController } from '../../controllers/ActionController';
-import { castToEditorResponse, getEditorResponseData } from '../../utils/EditorResponseData';
+import {ActionDeltaUpdate, ActionEditorEvent, ActionTrigger} from '../../types/ActionTypes';
+import {EditorAPI} from '../../types/CommonTypes';
+import {ActionController} from '../../controllers/ActionController';
+import {castToEditorResponse, getEditorResponseData} from '../../utils/EditorResponseData';
```
Comment (Matthiee) [inline packages/sdk/src/tests/controllers/ActionController.test.ts:4]: Fix formatting

**Review by Matthiee — CHANGES_REQUESTED** [review]

File: packages/sdk/src/controllers/ActionController.ts (hunk 3 above)
Comment (evan-mcgeek) [reply packages/sdk/src/controllers/ActionController.ts:3]: Prettier got broken again :(

File: packages/sdk/src/controllers/ActionController.ts [hunk 5]
```diff
@@ -139,4 +139,24 @@ export class ActionController {
...
+        const res = await this.#editorAPI;
+        return res.disableActions().then((result) => getEditorResponseData<null>(result));
+    };
+
+    /**
+     * This method enables the execution of Actions.
+     * Note this is enabled by default.
+     * @returns
+     */
+    enableExecution = async () => {
```
Comment (brapoprod) [inline packages/sdk/src/controllers/ActionController.ts:158]: just disable / enable? speaks for itself no?

Comment (NicolaVerbeeck) [reply packages/sdk/src/controllers/ActionController.ts:158]: Disable actioncontroller? Confusing

Comment (brapoprod) [reply packages/sdk/src/controllers/ActionController.ts:158]: in the end it becomes SDK.action.disable

Comment (NicolaVerbeeck) [reply packages/sdk/src/controllers/ActionController.ts:158]: Would love if it were actions, this now indicates 'disable a single action'. Though in this case disable makes sense indeed

**Review by brapoprod — APPROVED** [review]

**Pull request merged**